Machine_Learning/notebooks/czmodel/Regresssion_PyTorch_5_0_0.ipynb

```diff
@@ -0,0 +1,669 @@
+    "    compressed_data = './Mouse_Kidney_images.zip'\n",
+    "    if not os.path.isfile(compressed_data):\n",
+    "        import io\n",
+    "        response = requests.get(GITHUB_TRAINING_DATA_PATH, stream=True)\n",
+    "        compressed_data = io.BytesIO(response.content)\n",
+    "        \n",
+    "    import zipfile\n",
+    "    with zipfile.ZipFile(compressed_data, 'r') as zip_accessor:\n",
+    "        zip_accessor.extractall('./')\n",
+    "        \n",
+    "# Download model conversion spec\n",
+    "if not os.path.isfile('regression_conversion_spec_pytorch.json'):\n",
+    "    response = requests.get(GITHUB_MODEL_CONVERSION_SPEC_PATH, stream=True)\n",
+    "    with open('regression_conversion_spec_pytorch.json', 'wb') as handle:\n",
+    "        handle.write(response.content)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "h9B-lhwXrEEG"
+   },
+   "source": [
+    "### Define a PyTorch dataset to pre-process the images\n",
+    "Since the dataset contains very large images we need to train on smaller crops in order to not exhaust the GPU memory"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "hrIPBmrkrEEG"
+   },
+   "outputs": [],
+   "source": [
+    "from torch.utils.data import DataLoader, Dataset\n",
+    "import cv2\n",
+    "\n",
+    "# Dataset class\n",
+    "class SampleDataset(Dataset):\n",
+    "    def __init__(self, images_folder:str, labels_folder:str, transforms=None):\n",
+    "        self.sample_images = sorted([os.path.join(images_folder, f) for f in os.listdir(images_folder) \n",
+    "                                    if os.path.isfile(os.path.join(images_folder, f))])\n",
+    "        self.sample_labels = sorted([os.path.join(labels_folder, f) for f in os.listdir(labels_folder) \n",
+    "                                    if os.path.isfile(os.path.join(labels_folder, f))])\n",
+    "        self.transforms = transforms\n",
+    "\n",
+    "    def __getitem__(self, idx):\n",
+    "        image = self.sample_images[idx]\n",
+    "        label = self.sample_labels[idx]\n",
+    "        img = cv2.imread(image, cv2.IMREAD_GRAYSCALE)\n",
+    "        labl = cv2.imread(label, cv2.IMREAD_GRAYSCALE) \n",
+    "\n",
+    "        if self.transforms is not None:\n",
+    "            img = self.transforms(img)\n",
+    "            labl = self.transforms(labl)\n",
+    "\n",
+    "        return img.to(torch.float32), labl.to(torch.float32)\n",
+    "\n",
+    "    def __len__(self):\n",
+    "        return len(self.sample_images)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define transforms\n",
+    "from torchvision import transforms\n",
+    "transforms = transforms.Compose([\n",
+    "    transforms.ToTensor(),\n",
+    "    transforms.CenterCrop(CROP_SIZE),\n",
```

Comment:
AFAIR we are using random crop in the TensorFlow version. Is there a reason to use center crop here instead?